### PR TITLE
Platui 70 dynamic iframe height

### PR DESCRIPTION
**Pull request by silasl**
* Added `iframe-resizer` dependency
* Compile with site JS and apply to all iFrames on page load 
* 2 x Integration tests

### CODE REVIEW

File: application/__tests__/example.macro.integration.js [hunk 1]
```diff
@@ -2,12 +2,23 @@
 
 const { integrationTestPort } = require('../../constants')
 
-describe('Localhost', () => {
+describe('Example page rendering', () => {
   beforeEach(async () => {
-    await page.goto(`http://localhost:${integrationTestPort}`)
+    await page.goto(`http://localhost:${integrationTestPort}/design-library/ask-for-consent/`)
```
Comment (matcarey) [inline application/__tests__/example.macro.integration.js]: Should we do a full:

```
const baseUrl=`http://localhost:${integrationTestPort}`
```

That way we can use:

```
await page.goto(`${baseUrl}/design-library/ask-for-consent/`)
```

Or we could wrap it all up in:

```
const visit = url => page.goto(`http://localhost:${integrationTestPort}${url}`)

await visit('/design-library/ask-for-consent/`)
```

File: application/__tests__/example.macro.integration.js [hunk 2]
```diff
@@ -2,12 +2,23 @@
...
+    expect(title).toMatch('Ask the user for their consent')
+  })
+
+  test('iframes should have a style attribute with a height property', async () => {
+    const noHeightProperty = await page.$$eval('iframe', iframes => iframes.filter(iframe => !iframe.style.height))
+    expect(noHeightProperty).toHaveLength(0)
+  })
+
+  test('iframes should be resized to display all content', async () => {
+    const resizeFailed = await page.$$eval('iframe', iframes => iframes.filter(iframe => iframe.offsetHeight < iframe.contentWindow.document.body.offsetHeight))
```
Comment (matcarey) [inline application/__tests__/example.macro.integration.js]: I'd put this into a function, just to give it a name:

```
iframes.filter(isFrameATardis)
```

You'd want a better name than that, it's a function which returns true if the frame is bigger on the inside so we could call it `isFrameContentBiggerThanFrameItself` etc.

File: gulpfile.js [hunk 3]
```diff
@@ -9,9 +9,9 @@ require('./tasks/gulp/sass')
 require('./tasks/gulp/serve')
 require('./tasks/gulp/jest')
 
-gulp.task('watch', ['build:watch', 'scss:watch', 'copy-assets:watch', 'jest:unit:watch'])
+gulp.task('watch', ['build:watch', 'scss:watch', 'copy-assets:watch'])
 
-gulp.task('build', ['clean', 'compile', 'scss:compile', 'copy-assets'])
+gulp.task('build', ['lint', 'clean', 'compile', 'scss:compile', 'copy-assets'])
```
Comment (matcarey) [inline gulpfile.js]: Hmm, I'm not convinced by this.  I'd personally run: `npm test && npm run build`, in a number of projects I've actually found it a problem when you can't build with failing stuff - mostly for local testing.

File: package.json [hunk 4]
```diff
@@ -32,7 +32,8 @@
   "homepage": "https://github.com/hmrc/design-system#readme",
   "dependencies": {
     "govuk-frontend": "^2.11.0",
-    "hmrc-frontend": "^0.5.0"
+    "hmrc-frontend": "^0.5.0",
+    "iframe-resizer": "^4.1.1"
```
Comment (matcarey) [inline package.json]: I'd say this is a `devDependency` as it's static site generated ... `govuk-frontend` and `hmrc-frontend` could both be `devDependency` too but this sets them apart in a way that I find helpful.

File: application/__tests__/example.macro.integration.js (hunk 1 above)
Comment (silasl) [reply application/__tests__/example.macro.integration.js]: 👍

File: application/__tests__/example.macro.integration.js (hunk 2 above)
Comment (silasl) [reply application/__tests__/example.macro.integration.js]: 👍 Tardis name retained due to outstanding clarity

File: gulpfile.js (hunk 3 above)
Comment (silasl) [reply gulpfile.js]: 👍

File: package.json (hunk 4 above)
Comment (silasl) [reply package.json]: 👍

File: application/__tests__/example.macro.integration.js [hunk 5]
```diff
@@ -1,13 +1,28 @@
...
+  })
+
+  test('iframes should have a style attribute with a height property', async () => {
+    const noHeightProperty = await page.$$eval('iframe', iframes => iframes.filter(iframe => !iframe.style.height))
+    expect(noHeightProperty).toHaveLength(0)
+  })
+
+  test('iframes should be resized to display all content', async () => {
+    const resizeFailed = await page.$$eval('iframe', (iframes) => {
+      const isFrameATardis = iframe => iframe.offsetHeight < iframe.contentWindow.document.body.offsetHeight
```
Comment (matcarey) [inline application/__tests__/example.macro.integration.js]: Could this not be refactored into a function?

**Pull request merged**